Update Crypto Algorithm for May,2021 HUST School network update

Gaomengkai opened the pull request:
just some small fix, but I use `rsa` instead of `Crypto`

Gaomengkai committed:
update crypto

Gaomengkai committed:
Delete debug prints

Cra1gTucker commented:
I have noticed that you changed the log in test URL to `192.168.50.1`. Are you sure this would work across the whole campus? If you intended to get the redirect info by accessing the network portal on that IP, then notice that the portal address is different in different areas of the campus (a different IP address starting with 172 in some places). Also `123.123.123.123` is the default redirection you'll get when you try to visit the internet not logged in. So unless you're sure that this change works everywhere, I suggest not to touch it. Plus I don't think this script should consider users using a proxy (why not start the proxy after logging in?).

Cra1gTucker committed:
small changes to deal with warnings

Also changed the detection URL back to 123.123.123.123 (see reason in PR reply). May change the URL to an online portal detection URL in the future.

Cra1gTucker commented:
Merging request first, testing is still required.
